src/python/pants/backend/go/util_rules/third_party_pkg.py

```diff
@@ -441,11 +446,20 @@ async def analyze_go_third_party_package(
         imports=tuple(request.pkg_json.get("Imports", ())),
         go_files=tuple(request.pkg_json.get("GoFiles", ())),
         s_files=tuple(request.pkg_json.get("SFiles", ())),
+        cgo_files=tuple(request.pkg_json.get("CgoFiles", ())),
         minimum_go_version=request.minimum_go_version,
         embed_patterns=tuple(request.pkg_json.get("EmbedPatterns", [])),
         test_embed_patterns=tuple(request.pkg_json.get("TestEmbedPatterns", [])),
         xtest_embed_patterns=tuple(request.pkg_json.get("XTestEmbedPatterns", [])),
         error=maybe_error,
+        cgo_flags=CGoCompilerFlags(
+            cflags=tuple(request.pkg_json.get("CgoCFLAGS", [])),
+            cppflags=tuple(request.pkg_json.get("CgoCPPFLAGS", [])),
+            cxxflags=tuple(request.pkg_json.get("CgoCXXFLAGS", [])),
+            fflags=tuple(request.pkg_json.get("CgoFFLAGS", [])),
+            ldflags=tuple(request.pkg_json.get("CgoLDFLAGS", [])),
+            pkg_config=tuple(request.pkg_json.get("CgoPkgConfig", [])),
+        ),
     )
```

Comment:
Thoughts on de-duplicating some of this between `first_party_pkg.py` and this file? This is probably fine already since things are so stable.

Reply:
https://github.com/pantsbuild/pants/issues/16832